tests/api/v0/utils/security/test__security.py

```diff
@@ -0,0 +1,132 @@
+import pytest
+from datetime import datetime
+import os
+from hashlib import sha3_512
+
+from sqlalchemy import create_engine
+from sqlalchemy import MetaData, Table, Column, Integer, String, DateTime
+from sqlalchemy.orm import Session
+from sqlalchemy.exc import NoResultFound, MultipleResultsFound
+
+from api.v0.utils.security import _security
+from api.v0.utils.security import _user
+
+@pytest.fixture(scope="function")
+def engine(monkeypatch):
+    return create_engine("sqlite+pysqlite:///:memory:", echo=True)
+
+@pytest.fixture(scope="function")
+def metadata(engine):
+    return MetaData()
+
+@pytest.fixture(scope="function")
+def users_table(metadata):
+    return Table(
+        'users',
+        metadata,
+        Column('_id', Integer, primary_key=True),
+        Column('name', String(50), nullable=False, unique=True),
+        Column('password', String(128), nullable=False),
+        Column('salt', String(50), nullable=False),
+        Column('createdAt', DateTime, nullable=False),
+        Column('lastEdited', DateTime, nullable=False),
+    )
+
+@pytest.fixture(scope="function")
+def session(engine):
+    with Session(engine) as session:
+        yield session
+
+@pytest.fixture(scope="function")
+def put_users_table(metadata, engine, users_table):
+    try:
+        metadata.drop_all(engine, [users_table])
+    except:
+        pass
+
+    metadata.create_all(engine, [users_table])
+
+@pytest.fixture(scope="function")
+def mock_engine(engine, monkeypatch):
+    monkeypatch.setattr(_security, "engine", engine)
```

Comment:
Pull all this into a conftest.py. Similar stuff in `test__user.py`.

The two use different scopes -- perfectly fine to just go for `function` everywhere though. Not many tests; doubt this will go very slowly

Reply:
Mentioned in #14